### PR TITLE
PWN-7623 - claim interactor / repo / api - added to make claim and get fees

### PR DESCRIPTION
**Pull request by kabramovp2p**
## Jira Ticket

https://p2pvalidator.atlassian.net/browse/PWN-7623

## Description of Work

PWN-7623 - claim interactor / repo / api - added to make claim and get fees

### CODE REVIEW

File: app/src/main/java/org/p2p/wallet/claim/api/request/GetClaimFeeRpcRequest.kt [hunk 1]
```diff
@@ -21,5 +21,5 @@ class GetClaimFeeRpcRequest(
         put("amount", amount)
     }
 ) {
-    override val typeOfResult: Type = BundleFeeResponse::class.java
+    override val typeOfResult: Type = FeesResponse::class.java
```
Comment (gslevinkov) [inline app/src/main/java/org/p2p/wallet/claim/api/request/GetClaimFeeRpcRequest.kt:24]: there are a lot of Fees in our app
i would add some context
ClaimFeeResponse for exampe

File: app/src/main/java/org/p2p/wallet/claim/api/response/FeeResponse.kt [hunk 2]
```diff
@@ -0,0 +1,11 @@
+package org.p2p.wallet.claim.api.response
+
+import com.google.gson.annotations.SerializedName
+import java.math.BigDecimal
+import java.math.BigInteger
+
+class FeeResponse(
+    val amount: BigInteger,
```
Comment (gslevinkov) [inline app/src/main/java/org/p2p/wallet/claim/api/response/FeeResponse.kt:8]: amount is without SerializedName

File: app/src/main/java/org/p2p/wallet/claim/api/response/FeesResponse.kt [hunk 3]
```diff
@@ -0,0 +1,12 @@
+package org.p2p.wallet.claim.api.response
+
+import com.google.gson.annotations.SerializedName
+
+class FeesResponse(
+    @SerializedName("gas")
+    val gasFee: FeeResponse? = null,
```
Comment (gslevinkov) [inline app/src/main/java/org/p2p/wallet/claim/api/response/FeesResponse.kt:7]: FeeDetailsResponse

File: app/src/main/java/org/p2p/wallet/claim/interactor/ClaimInteractor.kt [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+import org.p2p.wallet.infrastructure.network.provider.TokenKeyProvider
+
+class ClaimInteractor(
+    private val ethereumClaimRepository: EthereumClaimRepository,
+    private val ethereumRepository: EthereumRepository,
+    private val tokenKeyProvider: TokenKeyProvider
+) {
+
+    suspend fun getClaimFees(
+        tokenAddress: String?,
```
Comment (gslevinkov) [inline app/src/main/java/org/p2p/wallet/claim/interactor/ClaimInteractor.kt:15]: tokenAddress is Hex, i think

File: app/src/main/java/org/p2p/wallet/claim/interactor/ClaimInteractor.kt [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+        amountAsString: String
```
Comment (gslevinkov) [inline app/src/main/java/org/p2p/wallet/claim/interactor/ClaimInteractor.kt:16]: asString is redundant - we see it's a string from type

File: app/src/main/java/org/p2p/wallet/claim/repository/EthereumClaimRemoteRepository.kt [hunk 6]
```diff
@@ -0,0 +1,25 @@
+package org.p2p.wallet.claim.repository
+
+import org.p2p.solanaj.model.types.RpcMapRequest
+import org.p2p.wallet.claim.api.BridgeApi
+import org.p2p.wallet.claim.api.response.FeesResponse
+
+class EthereumClaimRemoteRepository(private val api: BridgeApi) : EthereumClaimRepository {
+
+    override suspend fun getEthereumFees(
+        ethereumAddress: String,
```
Comment (gslevinkov) [inline app/src/main/java/org/p2p/wallet/claim/repository/EthereumClaimRemoteRepository.kt:10]: use wrapper object instead of primitives
thanks

**Review by gslevinkov — APPROVED** [review]

**Review by NikolaiEmelianov — APPROVED** [review]

File: app/src/main/java/org/p2p/wallet/claim/interactor/ClaimInteractor.kt (hunk 4 above)
Comment (kabramovp2p) [reply app/src/main/java/org/p2p/wallet/claim/interactor/ClaimInteractor.kt:15]: hex is String

File: app/src/main/java/org/p2p/wallet/claim/repository/EthereumClaimRepository.kt [hunk 7]
```diff
@@ -0,0 +1,12 @@
...
+
+import org.p2p.wallet.claim.api.response.FeesResponse
+
+interface EthereumClaimRepository {
+    suspend fun getEthereumFees(
+        ethereumAddress: String,
+        solanaAddress: String,
+        tokenAddress: String?,
+        amountAsString: String
+    ): FeesResponse
```
[the remark this answers is not in the document]
Comment (Davran331122) [reply app/src/main/java/org/p2p/wallet/claim/repository/EthereumClaimRepository.kt:11]: ethereumAddress == EthAddress, solanaAddress = SolAddress, erc20TokenAddress = EthAddress

File: app/src/main/java/org/p2p/wallet/claim/repository/EthereumClaimRemoteRepository.kt [hunk 8]
```diff
@@ -0,0 +1,25 @@
...
+        solanaAddress: String,
+        tokenAddress: String?,
+        amountAsString: String
+    ): FeesResponse {
+
+        val params = hashMapOf<String, Any?>(
```
Comment (Davran331122) [inline app/src/main/java/org/p2p/wallet/claim/repository/EthereumClaimRemoteRepository.kt:16]: You can move it to class, take  GetClaimFeeRpcRequst as an example

**Pull request merged**